core/http/endpoints/openai/chat.go

```diff
@@ -39,11 +39,15 @@ func ChatEndpoint(cl *config.BackendConfigLoader, ml *model.ModelLoader, startup
 		responses <- initialMessage
 
 		ComputeChoices(req, s, config, startupOptions, loader, func(s string, c *[]schema.Choice) {}, func(s string, usage backend.TokenUsage) bool {
+			choices := []schema.Choice{}
+			if s != "" {
```

Comment:
I'm not entirely sure about this as I didn't checked what OpenAI does in this case (and hard to replicate). 

If the token is empty does it send back an empty choice, or just skips it? Probably doesn't matter either, just making sure we keep an eye to not break any user flow

Reply:
i don't really know, but that token callback function is used only for streaming reply mode

and in that mode we chop runes of the token, and if there is some - we calling that function, so before that commit in that specific case the tokenCallback never will be called with empty first argument which i check here



the problem is the real usage is sent by llama as final reply where message field bytes is empty 

and in that specific case i explicitly call tokenCallback just to capture the usage part